vault/src/metrics.rs

```diff
@@ -428,48 +439,94 @@ async fn publish_issue_count<B: BitcoinCoreApi + Clone + Send + Sync>(
     }
 }
 
-async fn publish_redeem_count<B: BitcoinCoreApi + Clone + Send + Sync>(
+async fn publish_time_to_first_deadline<B: BitcoinCoreApi + Clone + Send + Sync>(
     parachain_rpc: &InterBtcParachain,
     vault_id_manager: &VaultIdManager<B>,
+    redeems: &[(H256, InterBtcRedeemRequest)],
 ) {
-    if let Ok(redeems) = parachain_rpc
-        .get_vault_redeem_requests(parachain_rpc.get_account_id().clone())
-        .await
-    {
-        for vault in vault_id_manager.get_entries().await {
-            let relevant_redeems: Vec<_> = redeems
+    for vault in vault_id_manager.get_entries().await {
+        let data: Result<_, crate::Error> = tokio::try_join!(
+            async { parachain_rpc.get_redeem_period().await.map_err(Into::into) },
+            async { parachain_rpc.get_current_chain_height().await.map_err(Into::into) },
+            async { vault.btc_rpc.get_block_count().await.map_err(Into::into) },
```

Comment:
I found something (just fyi): if you import use futures::TryFutureExt then instead of 
```
async { vault.btc_rpc.get_block_count().await.map_err(Into::into) },
````
you can do `vault.btc_rpc.get_block_count().map_err(Into::into)`